app/controllers/tasks_controller.rb

```diff
@@ -1,88 +1,70 @@
 class TasksController < ApplicationController
-  before_action :set_task, only: %i[show edit update destroy assign update_assign]
-  before_action :is_authorized_user?, only: %i[ edit update destroy assign update_assign]
+  before_action :set_task, only: %i[show]
+  before_action :set_mytask, only: %i[edit update destroy update_assign edit_assign]
   before_action :authenticate_user!
-  before_action :set_q, only: [ :index, :search, :mypage ]
+  before_action :set_q, only: %i[index mypage]
 
   def index
-    if params[:new]
-      @tasks = Task.latest.page(params[:page])
-    elsif params[:old]
-      @tasks = Task.old.page(params[:page])
-    elsif params[:emergency]
-      @tasks = Task.emergency.page(params[:page])
-    else
-     @tasks = Task.where.not(status:"complete").page(params[:page])
-    end
+    @tasks = Task.where.not(status: 'complete').preload(:user)
```

Comment:
scope使ってあげてもいいかもしれないです。
```suggestion
    @tasks = Task. incomplete.preload(:user)
```
```
  scope :incomplete, -> (status) {
    where(status: status || ["未対応", "対応中"])
  }
```